### DISCUSSION TITLE
sc68 files

### DISCUSSION THREAD

**Discussion by EarBiteR** [audacious-media-player/audacious#1468] [Q&A]
Does Audacious have the ability to play these old Atari ST and Amiga songs?  Supposedly there was a SC68 XMMS plugin at one time and Audacious is a port of that I thought.  

I found Linux and windows player for sndh and it it wouldve been great if Audacious supported that as well.

Just asking as I wouldnt expect anyone is really that interested in these formats from the last century!

**Comment by mschwendt** [comment]
Hmmm. Strictly speaking, SC68 is less "a file format" than it is about a rather complete emulator engine with specific capabilities as to emulate a set of chips used by the Atari ST (but not only that one). Sure, there are *.sc68 file, but the requirements to handle the code and data in those files are high. CPU emulation as to run native music player machine code plus emulation of audio chips and support chips. Because of that, SC68 is very exotic. Also, Atari ST music has had some fans, but AMIGA music has been much more popular.

There have been similar solutions, like DeliPlayer largely relying on emulation in addition to handling some file formats directly. For example, some music composers have used proprietary file formats and proprietary players, such as TFMX by Chris Hülsbeck and Peter Thierolf, and only by means of running the native player code (with bugs and glitches and all) and music data within an emulator, the resulting music would be played back accurately enough. Attempts at handling exotic file formats directly (perhaps only with emulated sound generation if not playing back samples only) may have been successful enough as to play a few files well enough while failing to handle many more. It's been similar with MOD files and the flood of derivatives.

I'm not familiar with the SC68 player that's available for Linux as source code. It seems to be unmaintained since 2003 (or a few years later). Interest in old compositions depends on how popular the music has been. That's why there are backend libraries for playing SID files, MOD files, FC files, and so on. An existing old plugin for XMMS wouldn't help much, since Audacious has changed a lot, and work would need to start with making available an SC68 emulator engine somehow, e.g. as a backend library. Not sure there's sufficient interest in that.

**Reply by EarBiteR** [reply]
Thanks for the reply  and explanation.  I was aware it would need emulated chips to work.  

I found https://bulba.untergrund.net/emulator_e.htm that I was using to play .sndh files and looked at sc68 but wasnt able to get it working under wine on linux. Same with rePlayer.  

I just gave deliplayer a try and sorry to report it fails as well as it cant find an output.  I will keep trying to get it going.

**Reply by mvtiaine** [reply]
For sndh there is also https://sndh.oxygenedemos.com/ (Web player) and https://www.creamhq.de/jam (Windows). IIRC WinJam should at least work under WINE. Also see https://sndh.atari.org/players.php

For Amiga tunes there is https://github.com/mvtiaine/audacious-uade ;)
(SC68 format not supported though)

**Comment by mvtiaine** [comment]
There is actually Audacious plugin included in SC68 sources, but it currently only supports Audacious <= 3.7 (before the latest plugin API change):
https://sourceforge.net/p/sc68/code/HEAD/tree/sc68-audacious/

I've added support for Audacious >= 3.8 API a few years ago, but haven't gotten around to finish it and submit a PR.

**Reply by EarBiteR** [reply]
I found BZR Player - http://bzrplayer.blazer.nu/versions.php that seems to work well on Linux under wine except for not detecting song length's on sc68's. It appears to be under active development and plays over 650 formats!  :-)

Could you post your work so I can see if I can build it and get it working with audacious?

**Reply by mvtiaine** [reply]
@EarBiteR ok here's the patch. It should apply against https://sourceforge.net/p/sc68/code/HEAD/tree/
cd sc68 && patch -p0 < sc68-audacious38.patch 

You should first build and install SC68 from root dir and then the Audacious plugin.
autoreconf -i && ./configure && make && make install
and then same inside sc68-audacious dir

The plugin does not have any settings, but IIRC it reads conf from /usr/local/share/sc68/sc68.cfg
[sc68-audacious38.patch](https://github.com/user-attachments/files/17193653/sc68-audacious38.patch)

It still works fine for me with Audacious 4.4.1, just the code needs some cleaning up (can't remember details anymore though :)

**Reply by mvtiaine** [reply]
Btw, here's list of retro related external Audacious input plugins I know about (that should still be compatible with current Audacious version). Haven't really tested most of them other than built+installed. Also some may conflict/overlap with others or Audacious internal plugins regarding formats supported.
https://github.com/mschwendt/xmp-plugin-audacious
https://github.com/vgmstream/vgmstream
https://gitlab.com/kode54/audacious-xsf
https://github.com/saschaklick/audacious-usf-plugin (x86 only)
https://github.com/ReimuNotMoe/Audacious-Plugin-VGS
https://sourceforge.net/p/sc68/code/HEAD/tree/sc68-audacious/ (with my patch)
https://github.com/mvtiaine/audacious-uade

**Reply by mschwendt** [reply]
There's also an Audacious plug-in for AMIGA Future Composer (= the format Jochen Hippel used originally):
https://github.com/mschwendt/libtfmxaudiodecoder
(edited to point at the very latest repo!)
